### PR TITLE
Model initialization

### PR DESCRIPTION
**Pull request by danielodeh**
GoTrade Model

### CODE REVIEW

File: app/Models/User.php [hunk 1]
```diff
@@ -59,8 +60,28 @@ class User extends Authenticatable
         'email_verified_at' => 'datetime'
     ];
     
+    /**
+     * @var
+     */
+    public $products;
```
Comment (afolabiolayinka) [inline app/Models/User.php]: Not necessary

File: app/Models/User.php [hunk 2]
```diff
@@ -59,8 +60,28 @@ class User extends Authenticatable
...
+
+    /**
+     * @var
+     */
+    public $cartItem;
+
     public function products()
     {
-        return $this->hasMany('App\Models\Product', 'own_by');
+        return $this->hasMany('App\Models\Product', 'id', 'own_by');
```
Comment (afolabiolayinka) [inline app/Models/User.php]: this is the other way around...

return $this->hasMany('App\Models\Product', 'own_by','id');

illustartion -> return $this->belongsTo('App\Models\Product', 'foreign_key', 'other_key');

Comment (afolabiolayinka) [reply app/Models/User.php]: and the only time you add 'other_key' or 'local_key' is if User has a different PKey other than ID

File: app/Models/State.php [hunk 3]
```diff
@@ -21,4 +21,34 @@ class State extends Model
...
+    {
+        return $this->belongsTo('App\Models\Country', 'country_id', 'id');
+    }
+
+    /**
+     * Relation with cities table.
+     *
+     * @return \Illuminate\Database\Eloquent\Relations\HasMany
+     */
+    public function city()
```
Comment (afolabiolayinka) [inline app/Models/State.php:40]: country is not directly linked to city

File: app/Models/CartItems.php [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+     * @var string
+     */
+    protected $table = 'cart_items';
+
+    /**
+     * The attributes that are mass assignable.
+     *
+     * @var array
+     */
+    protected $fillable = ['quantity'];
```
Comment (afolabiolayinka) [inline app/Models/CartItems.php]: more fields needing here

File: app/Models/CartItems.php [hunk 5]
```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Model;
+
+/**
```
Comment (afolabiolayinka) [inline app/Models/CartItems.php]: 7 to 16 not necessary

File: app/Models/CartItems.php [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+ * Class CartItems
```
Comment (afolabiolayinka) [inline app/Models/CartItems.php]: 8 to 16 not necessary

File: app/Models/Country.php [hunk 7]
```diff
@@ -21,4 +21,19 @@ class Country extends Model
     protected $fillable = [
         'name',
     ];
+
+     /**
+     * @var
+     */
+    public $state;
```
Comment (afolabiolayinka) [inline app/Models/Country.php]: not needed

File: app/Models/Country.php [hunk 8]
```diff
@@ -21,4 +21,19 @@ class Country extends Model
...
+
+    /**
+     * Relation with states table.
+     *
+     * @return \Illuminate\Database\Eloquent\Relations\HasMany
+     */
+    public function state()
```
Comment (afolabiolayinka) [inline app/Models/Country.php]: states() not state()

File: app/Models/State.php (hunk 3 above)
Comment (danielodeh) [reply app/Models/State.php:40]: I didn't make reference to your comment at all or maybe I'm not getting what you meant by this comment

**Review by afolabiolayinka — APPROVED** [review]
Great job

**Pull request merged**